build.sh

```diff
@@ -638,6 +638,10 @@ __msbuildonunsupportedplatform=0
 
 source "$__ProjectRoot"/_build-commons.sh
 
+if [ "${__BuildArch}" != "${__HostArch}" ]; then
+    __CrossBuild=1
```

Comment:
This block also needs to be copied to build-test.sh

Reply:
I moved it to `_build-commons.sh`.

Reply:
When `__CrossBuild` is set to `1` in test, the test run fails in CI with ([e.g.](https://dev.azure.com/dnceng/public/_build/results?buildId=422134)):

```
[command]/bin/bash --noprofile --norc /__w/_temp/ef9689e3-91fd-4eba-9416-b62d51e4beb1.sh
Error rootfsDir has been passed, but the location is not valid.
```

This error is coming from `init-distro-rid.sh`. I will keep it as it is in master for now.

Reply:
The same issue happens with build.sh if someone passed in the option to skip the native build (`-skipmanaged`) and wanted to build managed components for other architecture. It seems we should fix that (but not in this PR).